Python/flowgraph.c

```diff
@@ -1544,10 +1717,73 @@ optimize_if_const_binop(basicblock *bb, int i, PyObject *consts, PyObject *const
         PyErr_Clear();
         return SUCCESS;
     }
-    int newopcode, newoparg;
-    RETURN_IF_ERROR(newop_from_folded(newconst, consts, const_cache, &newopcode, &newoparg));
-    nop_out(bb, i-1, 2);
-    INSTR_SET_OP1(binop, newopcode, newoparg);
+    RETURN_IF_ERROR(instr_make_load_const(binop, newconst, consts, const_cache));
+    nop_out(bb, i-1, BINOP_OPERAND_COUNT);
+    return SUCCESS;
+}
+
+static PyObject*
+const_folding_unary_not(PyObject *v)
```

Comment:
This should be named `eval_unary_not`. But I'd actually inline it in `eval_const_unaryop`. It's only used there and it's not a large function.